src/components/marketplace/buy_sell.tsx

```diff
@@ -203,26 +251,59 @@ class BuySell extends React.Component<Props, State> {
                             <TokenText>wETH</TokenText>
                         </TokenContainer>
                     </FieldContainer>
-                    <Button theme="secondary" onClick={tab === Tab.Buy ? this.buy : this.sell}>
-                        {tab === Tab.Buy ? 'Buy' : 'Sell'} {selectedTokenSymbol}
+
+                    <LabelContainer>
+                        <Label>Order Details</Label>
+                        <InnerTabs tabs={ethUsdTabs} />
+                    </LabelContainer>
+                    <Row>
+                        <Label color={themeColors.textLight}>Fee</Label>
+                        <Value>
+                            {orderDetailType === OrderDetailsType.Usd
+                                ? `$ ${this.state.orderFeeDollar.toFixed(2)}`
+                                : `${this.state.orderFeeEther.toString()} Eth`}
+                        </Value>
+                    </Row>
+
+                    <Button theme="secondary" onClick={tab === OrderSide.Buy ? this.buy : this.sell}>
+                        {tab === OrderSide.Buy ? 'Buy' : 'Sell'} {selectedTokenSymbol}
                     </Button>
                 </Content>
             </BuySellWrapper>
         );
     };
 
-    public changeTab = (tab: Tab) => () => this.setState({ tab });
+    public changeTab = (tab: OrderSide) => () => this.setState({ tab }, () => this.updateOrderFee());
 
     public updateMakerAmount = (newValue: BigNumber) => {
-        this.setState({
-            makerAmount: newValue,
-        });
+        this.setState(
+            {
+                makerAmount: newValue,
+            },
+            () => this.updateOrderFee(),
+        );
     };
 
     public updatePrice: React.ReactEventHandler<HTMLInputElement> = e => {
         const price = parseFloat(e.currentTarget.value);
+        this.setState({ price }, () => this.updateOrderFee());
+    };
 
-        this.setState({ price });
+    public updateOrderFee = async () => {
+        const orderFeeEther = orderDetailsFeeEther(
+            this.state.makerAmount,
+            new BigNumber(this.state.price),
```

Comment:
Am I missing something or this function expects to receive the taker amount, not the price?

Reply:
The price is the wETH that the user expects to obtain from the sale. Both are ERC20 (ZRX is the makerAmount and the wETH is the takerAmoun). Please correct me if it's wrong.
Regards

Reply:
No, that's not how it works. The maker and taker amount are the things that are being traded. The price is the relation between them. The only case where the price is equal to the taker amount (in the case of a sell) is when you are selling exactly 1 token.